### PR TITLE
Update airbase to 155 and friends

### PR DESCRIPTION
**Pull request by wendigo**
This introduces following improvements:

- better support for JDK 21-23 due to upgraded maven plugins and dependencies (PMD finally working on JDK 21)
- [reproducible maven build support](https://maven.apache.org/guides/mini/guide-reproducible-builds.html) (out of the box)
- checkstyle improvements to allow for `unnamed variables & patterns` (see https://openjdk.org/jeps/443)
- updates to some dependencies

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -290,7 +290,7 @@
             <dependency>
                 <groupId>org.ow2.asm</groupId>
                 <artifactId>asm-bom</artifactId>
-                <version>9.7</version>
+                <version>${dep.asm.version}</version>
```
Comment (martint) [inline pom.xml:293]: Is this related or does it belong in a separate commit?

**Review by martint — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (wendigo) [reply pom.xml:293]: Doesn't belong here. Fixing

**Comment by findepi** [comment]
> * reproducible maven build support (out of the box)

❤️
